### PR TITLE
fix(modal): switch to dialog element

### PR DESCRIPTION
**Pull request by Leotheluck**
- `OsdsModal` is now using a `HTMLDialogElement` and is no longer appended to the body. (_fixes react children problem)_
- `OsdsModal` is now reactive to `Escape` key. _(will close the modal if dismissible)_
- `OsdsModal`'s icon has been fixed and no longer features a colored background.
- Tests have been added to check if focus is well applied.

### CODE REVIEW

File: packages/components/modal/src/components/osds-modal/osds-modal.spec.ts [hunk 1]
```diff
@@ -24,11 +23,14 @@ describe('spec:osds-modal', () => {
 
     page = await newSpecPage({
       components: [OsdsModal],
-      html: `<osds-modal ${odsStringAttributes2Str(stringAttributes)}>My Modal</osds-modal>`,
+      html: `<osds-modal ${odsStringAttributes2Str(stringAttributes)}>Modal</osds-modal>`,
     });
 
     root = page.root;
     instance = page.rootInstance;
+    instance.modal = {
```
Comment (aesteves60) [inline packages/components/modal/src/components/osds-modal/osds-modal.spec.ts:31]: Is it work ?

File: packages/components/modal/src/components/osds-modal/osds-modal.tsx [hunk 2]
```diff
@@ -56,40 +78,25 @@ export class OsdsModal implements OdsModalAttribute, OdsModalMethod, OdsModalEve
...
-
-  componentWillLoad(): void {
-    document.body.appendChild(this.el);
-    this.watchOpenedStateHandler(this.masked ?? false);
+  componentDidLoad(): void {
+    this.handleShownState();
   }
 
-  render() {
+  render(): void {
```
Comment (aesteves60) [inline packages/components/modal/src/components/osds-modal/osds-modal.tsx:85]: return JSX.Element

File: packages/components/modal/src/components/osds-modal/osds-modal.tsx [hunk 3]
```diff
@@ -56,40 +78,25 @@ export class OsdsModal implements OdsModalAttribute, OdsModalMethod, OdsModalEve
...
-        <div class="wrapper">
+        <dialog
+          class="wrapper"
+          ref={(el?: HTMLElement | null): void => {
+            this.modal = el as HTMLDialogElement;
+          }}>
           <div class="header">
             {dismissible && (
-              <osds-button onClick={() => this.close()} color={color} circle={true} ghost={true}>
+              <osds-button onClick={(): Promise<void> => this.close()} color={color} circle variant='ghost'>
```
Comment (aesteves60) [inline packages/components/modal/src/components/osds-modal/osds-modal.tsx:99]: Use enum for variant

File: packages/components/modal/src/components/osds-modal/styles/osds-modal.color.scss [hunk 4]
```diff
@@ -31,6 +29,10 @@
     .wrapper {
       background: #fff;
     }
+
+    .wrapper::backdrop {
```
Comment (aesteves60) [inline packages/components/modal/src/components/osds-modal/styles/osds-modal.color.scss:33]: use the scss syntaxe .wrapper { &::backdrop { } }

**Review by aesteves60 — CHANGES_REQUESTED** [review]

**Review by aesteves60 — APPROVED** [review]

**Review by skhamvon — APPROVED** [review]

**Review by astagnol — APPROVED** [review]

**Pull request merged**